assets/js/components/CookieOverlay.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const utils = require('../libs/elife-utils')();
+
+module.exports = class CookieOverlay {
+
+  constructor($elm, _window = window, doc = document) {
+    if (!$elm) {
+      return;
+    }
+
+    this.$elm = $elm;
+    this.window = _window;
+    this.doc = doc;
+
+    if (CookieOverlay.previouslyAccepted(this.doc)) {
+      return;
+    }
+
+    const message = 'This site uses cookies to deliver its services and analyse traffic. By using ' +
+                    'this site, you agree to its use of cookies. <a href="/privacy">Learn more</a>.';
+    this.$overlay = this.buildDOM(message);
+    this.doc.querySelector('body').appendChild(this.$overlay);
+  }
+
+  static previouslyAccepted(doc) {
+    return !!utils.getCookieValue('cookieNotificationAccepted', doc.cookie);
+  }
+
+  buildDOM(message) {
+    const $root = utils.buildElement('div', ['opt-in-message-bar']);
+    utils.buildElement('div', ['opt-in-message-bar__message'], message, $root);
+
+    const $buttonWrapper = utils.buildElement('div', ['opt-in-message-bar__button_wrapper'], '', $root);
+    const $button = utils.buildElement('button', ['button', 'button--default', 'button--small'], 'Got it', $buttonWrapper);
+
+    $button.addEventListener('click', () => {
+      this.$overlay.parentNode.removeChild(this.$overlay);
+      const expiryDate = 'Fri, 31 December 2038 23:59:59 GMT';
+      this.doc.cookie = `cookieNotificationAccepted=true; expires=${expiryDate} path=/; domain=elifesciences.org;`;
```

Comment:
Not going to work for some places (eg local development), but we should turn this off there anyway.

Reply:
Could lose the `domain` (`path` seems redundant too), but means having a popup for each app in each instance.

Reply:
`path` is needed so that someone landing on e.g. https://elifesciences.org/articles/[id], clicking 'Got it' and getting the cookie to hide the overlay, will still have the cookie when on a page that's not within `/articles/`.

Reply:
Thought the default was the root for some reason, but sensibly it's not.